### DISCUSSION TITLE
Self Hosting Repositories

### DISCUSSION THREAD

**Discussion by trymeouteh** [pacstall/pacstall#546] [Ideas]
The AUR is centralized and therefore those who control the AUR catalog can censor packages from the AUR repo. Please allow anyone to self host their own catalog of Pacstall packages which can be easily added/imported to ones package manager.

This is also useful if the developer of a package wants to self host their pacscripts and maintain them.

**Comment by wizard-28** [comment]
We already allow this. As stated in this feature bullet point in the readme.

> Ability to install programs from multiple repositories

We support GitHub and GitLab repositories at the moment. You can read the [instructions](https://github.com/pacstall/pacstall-programs#pacstall-programs) to know how to make your own.

We also support tracking pacstall updates from other Github/lab repositories.

**Reply by D-Brox** [reply]
Technically, we support any repo, as long as it follows the correct structure:
```console
package-repository/
├── packagelist
└── packages/
    └── example-package/
        └── example-package.pacscript
```
So any url should work if you have `${url}/packagelist` containing the list of packages and the proper pacscript paths `${url}/${pkgname}/${pkgname}.pacscript`

We do facilitate adding github and gitlab repos tho, since their urls need parsing for accessing the raw file.

**Reply by D-Brox** [reply]
Be aware that you need to add the url root directory to `/usr/share/pacstall/repo/pacstallrepo.txt`.

Just adding `https://github.com/pacstall/pacstall-repo` wouldn't work. But adding:

`https://raw.githubusercontent.com/pacstall/pacstall-programs/master`
or
`https://github.com/pacstall/pacstall-repo/raw/master`

would work, even tho both of those urls are technically invalid, since pacstall just uses them to parse valid urls.

**Reply by wizard-28** [reply]
We're improving the user experience for this in 2.0.0, we'll have a well documented `config.toml` file where the end users will be able to add the sources (amongst other settings).

**Comment by trymeouteh** [comment]
Are there any other 3rd party repos out there currently?

**Reply by Elsie19** [reply]
There is this: https://github.com/sarmong/pacstall-sarmong

This particular repo can be added with this command:
```bash
pacstall -A https://github.com/sarmong/pacstall-sarmong
```
